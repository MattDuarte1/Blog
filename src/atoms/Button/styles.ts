import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme, disabled }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 20px 23px 19px;
    gap: 10px;
    border: none;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};

    min-width: 145px;
    height: 67px;
    font-weight: 700;
    color: #ffff;

    background: ${theme.colors.blue};
    border-radius: 8px;
    transition: filter 0.4s ease;

    filter: ${disabled && 'brightness(0.7)'};

    &:hover {
      filter: ${!disabled && 'brightness(1.1)'};
    }
  `}
`;
