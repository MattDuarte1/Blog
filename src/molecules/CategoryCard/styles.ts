import styled, { css, DefaultTheme } from 'styled-components';

const MediaContainer = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      width: 80%;
    }
  `,
};

interface ContainerProps {
  isActive: boolean;
}

const MenuIsActive = (isActive: boolean, theme: DefaultTheme) =>
  isActive &&
  css`
    background-color: ${theme.colors.blue};
    color: #fff;
  `;

export const Container = styled.article<ContainerProps>`
  ${({ theme, isActive }) => css`
    display: flex;
    align-items: center;
    padding-left: ${theme.spacings.large};
    width: 220px;
    height: 280px;
    color: ${theme.colors.black};
    background-color: ${theme.colors.white};
    filter: drop-shadow(4px 6px 13px rgba(0, 0, 0, 0.1));
    border-radius: 8px;
    cursor: pointer;

    > div {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      text-align: center;
      padding: 0px;
      gap: 30px;
    }

    &:hover {
      background-color: ${theme.colors.blue};
      color: white;
    }

    ${MediaContainer.small(theme)}
    ${MenuIsActive(isActive, theme)}
  `}
`;
