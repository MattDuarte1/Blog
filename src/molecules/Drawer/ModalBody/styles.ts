import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    flex: 1 1 0%;
    display: flex;
    padding: 2rem;
    width: 100%;
    height: inherit;
    justify-content: flex-start;
    gap: 2rem;
    flex-direction: column;
    align-items: center;

    > a {
      min-width: 100%;

      &:hover {
        color: ${theme.colors.blue};
      }
    }
  `}
`;
