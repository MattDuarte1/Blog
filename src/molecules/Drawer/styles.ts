import styled, { css } from 'styled-components';

interface IModalProps {
  isOpen?: boolean;
}

export const ModalBackDrop = styled.div<IModalProps>`
  ${({ isOpen }) => css`
    position: fixed;
    right: 0px;
    top: 0px;
    width: 100vw;
    transform: translateX(${isOpen ? '0%' : '100%'});
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.48);
    z-index: 100;
    transition: all 0.8s ease-in-out;
  `}
`;

export const Modal = styled.div<IModalProps>`
  ${({ theme, isOpen }) => css`
    position: fixed;
    right: 1px;
    top: 0px;
    bottom: 0px;
    width: ${isOpen ? '285px' : '0px'};
    padding: 25px;
    height: inherit;
    background-color: ${theme.colors.white};
    z-index: 110;
    transition: all 1s ease;

    > svg {
      position: absolute;
      top: 10px;
      right: 2rem;

      font-size: ${theme.font.sizes.large};
      color: ${theme.colors.black};
      cursor: pointer;
      transition: all 0.5s ease;

      &:hover {
        color: ${theme.colors.blue};
        transform: scale(1.1);
      }
    }
  `}
`;
