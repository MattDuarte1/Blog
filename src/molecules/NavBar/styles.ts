import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    height: 44px;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 0px;
    gap: 50px;

    @media ${theme.media.lteLarge} {
      display: none;
    }
  `}
`;
