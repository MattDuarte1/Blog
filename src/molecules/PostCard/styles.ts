import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreapoints = {
  AuthorDescription: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        > img {
          width: 47px;
          height: 47px;
        }
      }
    `,
  },
};

export const Container = styled.article`
  ${({ theme }) => css`
    width: 300px;
    height: 425px;
    padding: 1.4rem;
    background: ${theme.colors.white};
    position: relative;

    box-shadow: 4px 6px 13px rgba(0, 0, 0, 0.1);
    border-radius: 12px;
    text-align: start;
    transition: all 0.2s ease;
    cursor: pointer;

    > h1 {
      margin-left: 1rem;
    }

    img {
      margin-bottom: 2rem;
    }

    &:hover {
      transform: scale(1.02);
      filter: drop-shadow(0 1.5px 2px rgba(0, 0, 0, 0.2));
    }
  `}
`;

export const AuthorContent = styled.div`
  ${({ theme }) => css`
    position: absolute;
    padding: 0 1.4rem;
    width: 100%;
    height: 57px;
    left: 0px;
    top: 348px;
    display: flex;
    color: ${theme.colors.black};
  `}
`;

export const AuthorDescription = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 90%;
    flex-direction: row;
    justify-content: center;
    padding: 0px;
    gap: 15px;
    color: ${theme.colors.black};

    ${MediaBreapoints.AuthorDescription['small'](theme)}
  `}
`;

export const Avatar = styled.div`
  overflow: hidden;
  border-radius: 50%;
  width: 50px;
  height: 50px;
`;

export const AuthorInfo = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 2px;
  gap: 4.2px;

  width: 184px;
  height: 46.2px;
`;

export const ViewBox = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  flex-flow: column nowrap;

  > svg {
    color: ${({ theme }) => theme.colors.black};
    font-size: ${({ theme }) => theme.font.sizes.medium};
  }

  > h1 {
    display: inline;
  }
`;
