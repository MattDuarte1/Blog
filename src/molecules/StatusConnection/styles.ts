import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    color: ${theme.colors.white};
    background-color: ${theme.colors.black};
    display: flex;
    padding: 1rem;
    align-items: center;
    justify-content: center;
    font-weight: 700;

    > p {
      font-size: ${theme.font.sizes.xsmall};
    }

    @media ${theme.media.lteMedium} {
      > p {
        font-size: 1.2rem;
      }
    }
  `}
`;

interface IBall {
  status?: string;
}

const ballColor = {
  connecting: '#F1C40F',
  connected: '#1efc1e',
  closed: '#BA1200',
};

export const Ball = styled.span<IBall>`
  ${({ status }) => css`
    height: 0.75rem;
    width: 0.75rem;
    margin-right: 1rem;
    background-color: ${ballColor[status]};
    border-radius: 99999px;
    animation: pulse 1s infinite;

    @keyframes pulse {
      0% {
        transform: scale(0.95);
        box-shadow: 0 0 0 0 ${ballColor[status]};
      }

      70% {
        transform: scale(1);
        box-shadow: 0 0 0 5px rgba(155, 155, 155, 0.1);
      }

      100% {
        transform: scale(0.95);
        box-shadow: 0 0 0 0 rgba(155, 155, 155, 0.1);
      }
    }
  `}
`;
