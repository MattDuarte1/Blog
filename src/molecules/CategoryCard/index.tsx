import { forwardRef } from 'react';
import * as Styled from './styles';
import { Icategory } from '@/interfaces/Category';

export interface CategoryCardProps extends Icategory {
  onClick: () => void;
  categorySelected?: string;
}

const CategoryCard = forwardRef<HTMLElement, CategoryCardProps>(
  ({ title, id, image, onClick, categorySelected, ...props }, ref) => {
    return (
      <Styled.Container
        onClick={onClick}
        aria-details={title}
        ref={ref}
        id={id}
        isActive={categorySelected === title}
        {...props}
      >
        <div>
          <img src={image.url} alt={image.alt} />
          <h1>{title}</h1>
        </div>
      </Styled.Container>
    );
  },
);

CategoryCard.displayName = 'CategoryCard component';

export default CategoryCard;
