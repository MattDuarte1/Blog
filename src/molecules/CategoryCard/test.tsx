import CategoryCard, { CategoryCardProps } from '.';
import { renderTheme } from '@/styles/render-theme';
import { screen } from '@testing-library/react';
import { theme } from '@/styles/theme';

describe('<CategoryCard />', () => {
  const handleSelectCategory = jest.fn();
  const Mockprops: CategoryCardProps = {
    id: '1',
    title: undefined,
    onClick: handleSelectCategory,
    categorySelected: null,
    image: {
      url: 'www.google.com',
      alt: 'javascript',
    },
  };
  it('Should render CategoryCard component when not selected', () => {
    renderTheme(<CategoryCard {...Mockprops} />);
    const category = screen.getByRole('article');

    expect(category).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: /javascript/i }),
    ).toBeInTheDocument();

    // test styles of component default
    expect(category).toHaveStyle({
      'background-color': theme.colors.white,
      color: theme.colors.black,
      width: '220px',
      height: '280px',
      'padding-left': theme.spacings.large,
      cursor: 'pointer',
      display: 'flex',
      filter: 'drop-shadow(4px 6px 13px rgba(0,0,0,0.1))',
      'border-radius': '8px',
    });

    expect(category).toHaveStyleRule('width', '80%', {
      media: theme.media.lteSmall,
    });
  });

  it('Should render CategoryCard with backgroundColor {theme.colors.blue} when category is selected', () => {
    const newMockWithCategorySelected: CategoryCardProps = {
      ...Mockprops,
      categorySelected: 'javascript',
      title: 'javascript',
    };
    renderTheme(<CategoryCard {...newMockWithCategorySelected} />);

    expect(screen.getByRole('article')).toBeInTheDocument();
    expect(screen.getByRole('article')).toHaveStyle({
      'background-color': theme.colors.blue,
      color: '#fff',
      width: '220px',
      height: '280px',
      'padding-left': theme.spacings.large,
      cursor: 'pointer',
      display: 'flex',
      filter: 'drop-shadow(4px 6px 13px rgba(0,0,0,0.1))',
      'border-radius': '8px',
    });
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<CategoryCard {...Mockprops} />);

    expect(container.firstElementChild).toMatchSnapshot();
  });
});
