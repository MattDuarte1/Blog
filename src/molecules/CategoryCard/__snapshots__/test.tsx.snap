// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCard /> Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 3.576rem;
  width: 220px;
  height: 280px;
  color: #2B2C34;
  background-color: #FFFFFE;
  -webkit-filter: drop-shadow(4px 6px 13px rgba(0,0,0,0.1));
  filter: drop-shadow(4px 6px 13px rgba(0,0,0,0.1));
  border-radius: 8px;
  cursor: pointer;
}

.c0 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: center;
  padding: 0px;
  gap: 30px;
}

.c0:hover {
  background-color: #6246EA;
  color: white;
}

@media (max-width:400px) {
  .c0 {
    width: 80%;
  }
}

<article
  aria-details="javascript"
  class="c0"
  id="1"
>
  <div>
    <img
      alt="javascript"
      src="www.google.com"
    />
    <h1>
      javascript
    </h1>
  </div>
</article>
`;
